client/my-sites/domains/components/domain-and-plan-package/navigation/index.jsx

```diff
@@ -6,10 +6,16 @@ import './style.scss';
 export default function DomainAndPlanPackageNavigation( props ) {
 	const translate = useTranslate();
 
+	// `goBackLink` will be either wp-admin or My Home, depending on the user's
+	// current admin interface preference.
+	// If unavailable, we fall back to the previous page in history.
 	const goBack = () => {
 		if ( props.goBackLink ) {
 			window.location.assign( props.goBackLink );
-		} else {
+			return;
+		}
+
+		if ( window.history.length > 1 ) {
```

Comment:
Also, it seems that this condition was redundant before, since we always had `goBackLink`, according to the old condition:

```
const goBackButtonProps = launchpadScreen === 'full'
	? {
			goBackText: translate( 'Next Steps' ),
			goBackLink: `/setup/${ siteIntent }/launchpad?siteSlug=${ selectedSiteSlug }`,
	  }
	: {
			goBackLink: `/home/${ selectedSiteSlug }`,
	  };
```

So, maybe we can just remove it?

UPDATE: now you added `this.props.preferredView === 'wp-admin`, so looks like, only now, [it can be undefined](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/sites/selectors/get-site-admin-url.ts#L20). But can it be?
AFAIU, if there is `this.props.preferredView`, then we should definitely have `getSiteOption( state, siteId, 'admin_url' )` from `getSiteAdminUrl`, or I understand incorrectly?

Reply:
Thanks for these points @nightnei. You're right that with the original logic, the `window.history.go(-1)` was redundant, because `goBackLink` was always populated by a string. As long as that holds true, after addressing the rest of your feedback, I agree it can be removed.

Looking at the new logic, you're also right that `wpAdminUrl` could be `null`. I think the current change is still mostly safe, because (as you pointed out) if we're able to declare `preferredView` as `wp-admin`,  then we should have no trouble retrieving the `wpAdminUrl`. Both of those selectors are relying on `siteId`, so if one succeeds, they should both succeed. If `preferredView` WAS `null`, we'd skip to the second path of the ternary, and the having `wpAdminUrl` be `null` doesn't matter. We'd be sending them to Calypso at that point.

But I hate that I used the word "should" just now. I don't see a scenario where `preferredView === 'wp-admin'` but `wpAdminUrl === null`, but better safe than sorry. I'm going to add a check of the `wpAdminUrl` value into that ternary so we can be sure it's populated before using for the `goBackLink` value.